File: internal/model/property.go
package model

import "time"

type Property struct {
	ID         string    `json:"id"`
	Address    string    `json:"address"`
	Type       string    `json:"type"`
	Bedrooms   int       `json:"bedrooms"`
	Area       float64   `json:"area"`
	RentAmount float64   `json:"rent_amount"`
	Status     string    `json:"status"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

File: web/src/tests/msw/factories/property.ts
import type { TProperty } from '@/domains/properties';

export const createMockProperty = (overrides: Partial<TProperty> = {}): TProperty => ({
  id: 'p1',
  address: '123 Main St',
  type: 'apartment',
  bedrooms: 2,
  rent_amount: 1500,
  status: 'available',
  created_at: '2026-01-01T00:00:00Z',
  updated_at: '2026-01-01T00:00:00Z',
  ...overrides,
});
